build.gradle

```diff
@@ -42,8 +43,16 @@ plugins {
 }
 
 try {
-  // apply this plugin in a try-catch block so that we can handle cases without .git directory
+  // apply these plugins in a try-catch block so that we can handle cases without .git directory
   apply plugin: 'com.palantir.git-version'
+  apply plugin: 'com.gorylenko.gradle-git-properties'
+  // git properties file for the root project for adding to the source tarball
+  gitProperties {
+    gitPropertiesName = 'iceberg-build.properties'
+    extProperty = 'gitProps'
+    failOnNoGitDirectory = false
+  }
+  generateGitProperties.outputs.upToDateWhen { false }
```

Comment:
This file goes to `file("${rootDir}/build/resources/main/iceberg-build.properties`)`. 

It also saves all of the git properties onto the root projects `extensions` so that a write task could be written that uses them that is distinct from `generateGitProperties` if so desired.